### PR TITLE
BaSP-TG-32: tasks unit test

### PR DESCRIPTION
**Pull request by DinoStampella**

### CODE REVIEW

**Review by cami-diaz — APPROVED** [review]

File: src/tests/tasks.test.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      });
+    });
+  });
+  describe('Failed cases:', () => {
+    test('Should return status code 404 (bad request)', async () => {
+      const response = await request(app).get('/api/task').send();
+
+      expect(response.status).toBe(404);
+    });
+    test('Should retorun status code 400 (incorrect body)', async () => {
```
Comment (JulianFloresDev) [inline src/tests/tasks.test.js:57]: I think you could be more descriptive in what the test is intended to do..... When should return status code 400 ?

File: src/tests/tasks.test.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    test('Retourned should have fiteled value', async () => {
+      const response = await request(app).get('/api/tasks').send(filterByDescription);
+
+      response.body.data.forEach((task) => {
+        expect(task.description).toBe('BE');
+      });
+    });
+  });
+  describe('Failed cases:', () => {
+    test('Should return status code 404 (bad request)', async () => {
```
Comment (JulianFloresDev) [inline src/tests/tasks.test.js:52]: I think you could be more descriptive in what the test is intended to do..... When should return status code 404 ?

File: src/tests/tasks.test.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      const response = await request(app).get('/api/task').send();
+
+      expect(response.status).toBe(404);
+    });
+    test('Should retorun status code 400 (incorrect body)', async () => {
+      const response = await request(app).get('/api/tasks').send(falseFilterByDescription);
+
+      expect(response.status).toBe(400);
+    });
+    test('Should return Error: true', async () => {
```
Comment (JulianFloresDev) [inline src/tests/tasks.test.js:62]: Same as above

File: src/tests/tasks.test.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+describe('CREATE /tasks', () => {
+  const newTask = {
+    description: 'BE',
+  };
+  const falseNewTask = {
+    description: 'EE',
+  };
+  describe('Successful cases:', () => {
+    test('Should return status code 201', async () => {
```
Comment (JulianFloresDev) [inline src/tests/tasks.test.js:78]: I suggest bemor specific with test description.

File: src/tests/tasks.test.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+      const response = await request(app).post('/api/tasks').send(newTask);
+      expect(response.status).toBe(201);
+    });
+    test('New task should have the same content of the request body', async () => {
```
Comment (JulianFloresDev) [inline src/tests/tasks.test.js:82]: It is descriptive and specific with what is expected and what are the conditions that generate the expected condition.

That's what i mean!

**Comment by JulianFloresDev** [comment]
I suggest giving a little more description to the tests that are intended to be tested for better suit of testing practices; since the tests should be descriptive of the situation that is intended to be evaluated.

File: src/tests/tasks.test.js (hunk 1 above)
Comment (DinoStampella) [reply src/tests/tasks.test.js:57]: When you send wrong body (it's in parentheses)

File: src/tests/tasks.test.js (hunk 2 above)
Comment (DinoStampella) [reply src/tests/tasks.test.js:52]: When you make a bad request (it's in parentheses)

File: src/tests/tasks.test.js (hunk 3 above)
Comment (DinoStampella) [reply src/tests/tasks.test.js:62]: You're right, I forgot it

File: src/tests/tasks.test.js (hunk 4 above)
Comment (DinoStampella) [reply src/tests/tasks.test.js:78]: Return a status code 201 becouse is a successful case

File: src/tests/tasks.test.js (hunk 5 above)
Comment (DinoStampella) [reply src/tests/tasks.test.js:82]: I get you

File: src/tests/tasks.test.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+      const response = await request(app).post('/api/tasks').send(newTask);
+
+      expect(response.body.data.description).toBe(newTask.description);
+    });
+    test('Should return a success message', async () => {
```
[the remark this answers is not in the document]
Comment (DinoStampella) [reply src/tests/tasks.test.js:87]: When the request is successful

File: src/tests/tasks.test.js (hunk 2 above)
Comment (GuidoG95) [reply src/tests/tasks.test.js:52]: I think bad request should be status 400, 404 is for not found task

**Review by ignacioaristo — APPROVED** [review]

**Pull request merged**